src/server/services/wms/qgswmsrestorer.cpp

```diff
@@ -92,17 +78,12 @@ QgsLayerRestorer::~QgsLayerRestorer()
     layer->setName( mLayerSettings[layer].name );
 
     // if a SLD file has been loaded for rendering, we restore the previous style
-    if ( layer->customProperty( "readSLD", false ).toBool() )
+    const QString sldStyleName { layer->customProperty( "sldStyleName", "" ).toString() };
+    if ( !sldStyleName.isEmpty() )
     {
-      QString errMsg;
-      QDomElement root = settings.mQgisStyle.documentElement();
-      QgsReadWriteContext context = QgsReadWriteContext();
-      if ( !layer->readStyle( root, errMsg, context ) )
-      {
-        QgsMessageLog::logMessage( QStringLiteral( "QGIS Style has not been read from layer restorer for layer %1: %2" ).arg( layer->name(), errMsg ) );
-      }
+      layer->styleManager()->removeStyle( sldStyleName );
```

Comment:
I don't see an explicit style restoring here. The default style is reestablished as soon as the current style is removed?

Reply:
The style is restored earlier new line 77 old line 91
```py
layer->styleManager()->setCurrentStyle( settings.mNamedStyle );
```